File: leetcode/easy/find-target-indices-after-sorting-array.ts
// https://leetcode.com/problems/find-target-indices-after-sorting-array/

// MINE
/*
function targetIndices(nums: number[], target: number): number[] {
    const ans = []
    nums.sort((a, z) => a - z)
    for(let i=0; i<nums.length; ++i) if(nums[i] === target) ans.push(i)
    return ans
};
*/

// ANSWER: without sort
function targetIndices(nums: number[], target: number): number[] {
  let count = 0,
    less = 0
  const ans = []
  for (let n of nums) {
    if (n === target) ++count
    if (n < target) ++less
  }
  for (let i = 0; i < count; ++i) ans.push(less++)
  return ans
}

File: leetcode/easy/find-target-indices-after-sorting-array.py
# https://leetcode.com/problems/find-target-indices-after-sorting-array/

class Solution:
    def targetIndices(self, nums: List[int], target: int) -> List[int]:
        ''''
        count, less, ans = 0, 0, []
        for n in nums:
            if n == target:
                count += 1
            elif n < target:
                less += 1
        for i in range(count):
            ans.append(less)
            less += 1
        return ans
        '''

        # ANSWER
        lt_count, eq_count = 0, 0
        for n in nums:
            if n < target:
                lt_count += 1
            elif n == target:
                eq_count += 1

        return list(range(lt_count, lt_count+eq_count))
